MovieMVP/ServiceLayer/NetworkService.swift

```diff
@@ -0,0 +1,121 @@
+// NetworkService.swift
+// Copyright © RoadMap. All rights reserved.
+
+import Alamofire
+import Foundation
+import SwiftyJSON
+
+/// Протокол сетевого сервиса
+protocol NetworkServiceProtocol {
+    // MARK: - Public Methods
+
+    func fetchMovies(categoryMovies: String, completion: @escaping (Result<[Movie]?, Error>) -> Void)
+    func setupImageFromURLImage(posterPath: String, completion: @escaping (Result<Data?, Error>) -> Void)
+    func fetchSimilarMovies(idMovie: Int, completion: @escaping (Result<Data?, Error>) -> Void)
+}
+
+/// Сетевой сервис
+final class NetworkService: NetworkServiceProtocol {
+    // MARK: - Constants
+
+    private enum Constants {
+        static let apiKeyQueryText = "api_key=8216e974d625f2a458a739c20007dcd6"
+        static let languageQueryText = "&language=ru-RU"
+        static let pageQueryText = "&page=1"
+        static let themoviedbQueryText = "https://api.themoviedb.org/3/movie/"
+        static let posterPathQueryText = "https://image.tmdb.org/t/p/w500"
+        static let similarQueryText = "/similar?"
+        static let resultsText = "results"
+        static let posterPathText = "poster_path"
+        static let idText = "id"
+        static let overviewText = "overview"
+        static let releaseDateText = "release_date"
+        static let titleText = "title"
+        static let voteAverageText = "vote_average"
+        static let voteCountText = "vote_count"
+    }
+
+    // MARK: - Public Methods
+
+    func fetchMovies(categoryMovies: String, completion: @escaping (Result<[Movie]?, Error>) -> Void) {
+        let urlString = "\(Constants.themoviedbQueryText)\(categoryMovies)\(Constants.apiKeyQueryText)" +
+            "\(Constants.languageQueryText)\(Constants.pageQueryText)"
+        guard let url = URL(string: urlString) else { return }
+
+        AF.request(url, method: .get).responseData { response in
+            guard let safeData = response.value
+            else {
+                guard let safeError = response.error else { return }
+                completion(.failure(safeError))
+                return
+            }
```

Comment:
```suggestion
            guard let safeData = response.value
            else {
                guard let safeError = response.error else { return }
                completion(.failure(safeError))
                return
            }
```
тут switch просится, т.к. есть только два условия

Reply:
Не думаю что уже свитч, так как тут у меня проверка на гвардом на них